website/main/content/blog/2025-06-30-sector-expiration.md

```diff
@@ -0,0 +1,213 @@
+authors: [adlrocha]
+---
+
+As mentioned on last week's status update, one of the key pieces that I was to have the detailed
+operation of plot membership allocation was the impact of sector expiration on the protocol. By
+having a unique plot identifier, we are able to uniquely link sectors to plots, but these sectors
+need to expire in a way that does not require farmers to re-plot while archiving the most recent
+history. Fortunately, we can leverage the current expiration mechanism of the current Subspace
+protocol, and build a layer on top of it to adapt it to the sharded version while maintaining the
+original guarantees in terms of plot expiration, sector re-plotting, and history archiving.
+
+<!--more-->
+
+## Sharded Subspace Plot Expiration Protocol
+
+### Overview
+
+Before I start with the description of the protocol, let me remind you what are the core properties
+that we wanted the protocol to have:
+
+1.  **Unique Plot IDs:** `plot_id` is unique and stable. This is key for the protocol, as this ID
+    will be used to allocate space into shards.
+2.  **No Bias through `history_size`:** Farmers can't game shard allocation by choosing among any
+    `history_size`, or by creating sectors with the same ID committed to different history size.
+    This prevents an attack where the same plot ID could be used to create sectors with different
+    `history_size` values, which would allow farmers to manipulate shard allocation by assigning to
+    a plot the most convenient sector at each time.
+3.  **Load-Balanced Archiving (New History):** As new history arrives, it gets stored across
+    farmers/plots without large re-plotting events.
+4.  **Detect Expiration and Validity:** Plots, sectors and blocks include all the information needed
+    to verify their validity, plot assignment to a shard, sector belonging to the plot, and
+    expiration.
+5.  **No Excessive Re-plotting:** Keep re-plotting burden similar to original Subspace.
+
+### Committing sectors to history ranges
+
+The core idea for sector expiration is the following: instead of letting farmers commit their
+sectors to any history size, they are bound to commit them to a range of history sizes determined by
+the plot the sector is linked to. This way, we limit the range of history sizes that a sector can be
+committed to, and hence the parallel sectors with the same `plot_id` that and different committed
+history sizes that can be created. This is implemented as a "self-evolving plot commitment range"
+that grows as the blockchain grows on top of the existing mechanism for sector expiration. Thus, we
+maintain all the properties of the original Subspace protocol, while adapting it to the needs of the
+sharded version. This allows us to also learn an apply the heuristics from the live Subspace
+network.
+
+The high-level operation of the protocol is as follows:
+
+- When creating a new plot, farmers need to choose a random offset (essentially a nonce) for their
```

Comment:
I actually like "nonce" as a name for the parameters. Offset explains what it is, but "nonce" seems like a more established terminology for a parameter that user controls in various protocols.

Reply:
Agree. Renamed throughout the whole document